amcl/src/amcl_node.cpp

```diff
@@ -363,6 +365,15 @@ main(int argc, char** argv)
   return(0);
 }
 
+static rclcpp::Time rclcpp_now(std::shared_ptr<rclcpp::Node> node)
```

Comment:
Not sure that declaring rclcpp_now in leaf packages that used rclcpp::Time::now is the way to go.
Maybe @tfoote has a suggestion ? Is this how users should do in the future to be able to use a one-liner "time::now()" in their nodes?

Reply:
Yeah, I wasn't entirely sure about this.  I knew I didn't want to duplicate this code in all of the call sites below, but we may want to add a helper function within `rclcpp` itself.

Reply:
Actually, on second thought, maybe we want to create the `TimeSource` and the `Clock` in the constructor, and then just call `clock->now()` where we need it.  What do you think about that?

Reply:
yeah that sounds like a better alternative

Reply:
That's a much better approach. The timesource is subscribing and is designed to be long running in the background.